### PR TITLE
TrackFilesToNewArtifact operation

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: neptune/new/handler.py [hunk 1]
```diff
@@ -437,7 +436,7 @@ def fetch_files_list(self) -> List[ArtifactFileData]:
         """
         raise NeptuneException('Should be never called.')
 
-    def track_files(self, path: str) -> None:
+    def track_files(self, path: str, wait: bool = True) -> None:
```
Comment (Raalsky) [inline neptune/new/handler.py:439]: Add `namespace`

File: neptune/new/internal/backends/hosted_artifact_operations.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+def track_artifact_files(
+        backend: NeptuneBackend,
+        project_uuid: uuid.UUID,
+        path: List[str],
+        entries: List[Tuple[str, Optional[str]]]
+) -> Tuple[Optional[NeptuneException], Optional[Operation]]:
+    files: List[ArtifactFileData] = list()
+
+    for entry in entries:
+        entry_path, entry_namespace = entry
```
Comment (pkasprzyk) [inline neptune/new/internal/backends/hosted_artifact_operations.py:39]:
```suggestion
    for entry_path, entry_namespace in entries:
```

**Review by pkasprzyk — COMMENTED** [review]
looks great!

File: neptune/new/internal/backends/api_model.py [hunk 3]
```diff
@@ -183,7 +183,9 @@ class DatetimeAttribute:
 
 @dataclass
 class ArtifactAttribute:
```
Comment (pkasprzyk) [inline neptune/new/internal/backends/api_model.py:185]: this should model data stored in attribute API, so hash only

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 4]
```diff
@@ -619,6 +670,47 @@ def list_artifact_files(self, project_uuid: uuid.UUID, artifact_hash: str) -> Li
...
+    def create_new_artifact(self, project_uuid: uuid.UUID, artifact_hash: str, size: int) -> ArtifactAttribute:
+        params = {
+            'projectIdentifier': project_uuid,
+            'hash': artifact_hash,
+            'size': size,
+            **self.DEFAULT_REQUEST_KWARGS,
+        }
+        try:
+            result = self.artifacts_client.api.createNewArtifact(**params).response().result
+            return ArtifactAttribute(
```
Comment (pkasprzyk) [inline neptune/new/internal/backends/hosted_neptune_backend.py:683]: that's different object...

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 5]
```diff
@@ -600,7 +647,11 @@ def get_artifact_attribute(self, run_uuid: uuid.UUID, path: List[str]) -> Artifa
         }
         try:
             result = self.leaderboard_client.api.getArtifactAttribute(**params).response().result
-            return ArtifactAttribute(result.hash)
+            return ArtifactAttribute(
```
Comment (pkasprzyk) [inline neptune/new/internal/backends/hosted_neptune_backend.py:650]: ...than this one

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 6]
```diff
@@ -619,6 +670,47 @@ def list_artifact_files(self, project_uuid: uuid.UUID, artifact_hash: str) -> Li
...
+            'artifactFilesDTO': {
+                'files': [
+                    ArtifactFileData.to_dto(a) for a in files
+                ]
+            },
+            **self.DEFAULT_REQUEST_KWARGS
+        }
+        try:
+            result = self.artifacts_client.api.uploadArtifactFilesMetadata(**params).response().result
+            return ArtifactAttribute(
```
Comment (pkasprzyk) [inline neptune/new/internal/backends/hosted_neptune_backend.py:706]: same here, this is Artifact API response

**Review by pkasprzyk — CHANGES_REQUESTED** [review]

**Review by pkasprzyk — APPROVED** [review]

**Pull request merged**